### ISSUE TITLE
-i / --insecure flag does not permit insecure SSL/TLS connections

### ISSUE THREAD

**Issue by neolead** [bcoles/jira_scan#1]
```/usr/local/rvm/gems/ruby-2.6.3@metasploit-framework-master/gems/jira_scan-0.0.3/lib/jira_scan.rb:381: warning: instance variable
 @insecure not initialized
- Error: Could not retrieve URL https://jirabi.xxxxxxxxxxxxxxa.com/secure/Dashboard.jspa
SSL_connect returned=1 errno=0 state=error: certificate verify failed (unable to get local issuer certificate)
Traceback (most recent call last):
        3: from bin/jira-scan:180:in `<main>'
        2: from bin/jira-scan:93:in `scan'
        1: from /usr/local/rvm/gems/ruby-2.6.3@metasploit-framework-master/gems/jira_scan-0.0.3/lib/jira_scan.rb:41:in `detectJi
raDashboard'
/usr/local/rvm/gems/ruby-2.6.3@metasploit-framework-master/gems/jira_scan-0.0.3/lib/jira_scan.rb:401:in `sendHttpRequest': undef
ined method `body' for nil:NilClass (NoMethodError)
```

fixed with changing
```
http.verify_mode = @insecure ? OpenSSL::SSL::VERIFY_NONE : OpenSSL::SSL::VERIFY_PEER
http.verify_mode = OpenSSL::SSL::VERIFY_NONE
```

bash command:

cat /var/lib/gems/2.3.0/gems/jira_scan-0.0.3/lib/jira_scan.rb|sed 's/@insecure ? //g'|sed 's/ : OpenSSL::SSL::VERIFY_PEER//g' >/tmp/tmpjira; mv /tmp/tmpjira /var/lib/gems/2.3.0/gems/jira_scan-0.0.3/lib/jira_scan.rb

**Comment by bcoles** [comment]
Thanks. Yeah it's a bug. I'll have this fixed shortly.

**Comment by neolead** [comment]
> Thanks. Yeah it's a bug. I'll have this fixed shortly.

i think best practice is just totally ignore "secure" of ssl... and kill the key insecure

**Comment by bcoles** [comment]
> 
> 
> > Thanks. Yeah it's a bug. I'll have this fixed shortly.
> 
> i think best practice is just totally ignore "secure" of ssl... and kill the key insecure

That was the behaviour in 0.0.2 prior to 0.0.3 release.

That is easiest but far from best practice.

**Comment by neolead** [comment]
> > > Thanks. Yeah it's a bug. I'll have this fixed shortly.
> > 
> > 
> > i think best practice is just totally ignore "secure" of ssl... and kill the key insecure
> 
> That was the behaviour in 0.0.2 prior to 0.0.3 release.
> 
> That is easiest but far from best practice.

Usually, for scanning for vulnerabilities and leaks, especially if you are scanning an IP and not a hostname, you are just searching and manualling how to turn off the fucking ssl check =)

**Comment by neolead** [comment]
tens of python and other was modified by me to bypass ssl check.... its terrible... ...
example.. you route script over burp, to analize and modify... but...no... ssl
you use a over cloudflare direct ip... ssl error...fuuuuuuuuukcccc

**Comment by neolead** [comment]
if any, its my imho.

**Comment by bcoles** [comment]
This should now be fixed in version 0.0.4.

The `-i` flag will still need to be specified when scanning hosts which present an invalid/untrusted certificate.

**Comment by neolead** [comment]
> This should now be fixed in version 0.0.4.
> 
> The `-i` flag will still need to be specified when scanning hosts which present an invalid/untrusted certificate.

Confirmed fix
